### PR TITLE
HV-1062 Support property rule and type use rule together on a Map

### PR DESCRIPTION
**Pull request by gsmet**
https://hibernate.atlassian.net/browse/HV-1062

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java [hunk 1]
```diff
@@ -278,8 +278,8 @@ private UnwrapMode unwrapMode(ExecutableElement executable, boolean typeArgument
 		return constraints;
 	}
 
-	private UnwrapMode unwrapMode(boolean typeArgumentAnnotated, boolean notIterable, UnwrapValidatedValue unwrapValidatedValue) {
-		if ( unwrapValidatedValue == null && typeArgumentAnnotated && notIterable ) {
+	private UnwrapMode unwrapMode(boolean typeArgumentAnnotated, boolean notIndexable, UnwrapValidatedValue unwrapValidatedValue) {
+		if ( unwrapValidatedValue == null && typeArgumentAnnotated && notIndexable ) {
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java]: Can we make it a "positive" variable name? It's easier to grok that way, and more idiomatic: `!indexable`.

File: engine/src/main/java/org/hibernate/validator/internal/util/ReflectionHelper.java [hunk 2]
```diff
@@ -253,6 +253,28 @@ public static Object getValue(Method method, Object object) {
 	}
 
 	/**
+	 * Indicates if the type is considered indexable (ie is an <code>Iterable</code>, an array or a <code>Map</code>).
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/util/ReflectionHelper.java]: Could you use `{@code ...}` over `<code>...</code>` (as per http://hibernate.org/validator/contribute/)? If it's the other style still in some places, that's left-overs from earlier times (fix stuff you see as you go, if you like).

**Comment by gunnarmorling** [comment]
@gsmet Nice fix. Added two minor comments inline.

**Comment by gsmet** [comment]
@gunnarmorling I addressed your comments.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks, @gsmet!

**Pull request closed without merging**